projects/ngx-speculoos/src/lib/matchers.spec.ts

```diff
@@ -49,63 +49,105 @@ describe('Custom matchers', () => {
     jasmine.addMatchers(speculoosMatchers);
   });
 
-  it('should check for a class', () => {
-    expect(tester.div).toHaveClass('foo');
-    expect(tester.div).not.toHaveClass('baz');
-
+  describe('toHaveClass', () => {
     const matcher = speculoosMatchers.toHaveClass(undefined, undefined);
 
-    // missing class
-    let result = matcher.compare(tester.div, 'baz');
-    expect(result.pass).toBeFalsy();
-    expect(result.message).toBe(`Expected element to have class 'baz', but had 'foo, bar'`);
-
-    result = matcher.negativeCompare(tester.div, 'foo');
-    expect(result.pass).toBeFalsy();
-    expect(result.message).toBe(`Expected element to not have class 'foo', but had 'foo, bar'`);
-
-    // no class
-    result = matcher.compare(tester.none, 'baz');
-    expect(result.pass).toBeFalsy();
-    expect(result.message).toBe(`Expected element to have class 'baz', but had none`);
-
-    // null element
-    result = matcher.compare(null, 'baz');
-    expect(result.pass).toBeFalsy();
-    expect(result.message).toBe(`Expected to check class 'baz' on element, but element was falsy`);
-
-    // not a TestElement
-    result = matcher.compare('hello', 'baz');
-    expect(result.pass).toBeFalsy();
-    expect(result.message).toBe(`Expected to check class 'baz' on element, but element was not a TestElement`);
-  });
-
-  it('should check for a value', () => {
-    expect(tester.name).toHaveValue('Hello');
-    expect(tester.name).not.toHaveValue('baz');
+    it('should check for a class', () => {
+      expect(tester.div).toHaveClass('foo');
+      expect(tester.div).not.toHaveClass('baz');
+    });
+
+    it('should return false if class is missing', () => {
+      // missing class
+      let result = matcher.compare(tester.div, 'baz');
+      expect(result.pass).toBeFalsy();
+      expect(result.message).toBe(`Expected element to have class 'baz', but had 'foo, bar'`);
+
+      result = matcher.negativeCompare(tester.div, 'foo');
+      expect(result.pass).toBeFalsy();
+      expect(result.message).toBe(`Expected element to not have class 'foo', but had 'foo, bar'`);
+
+    });
+
+    it('should handle no class', () => {
+      // no class
+      let result = matcher.compare(tester.none, 'baz');
+      expect(result.pass).toBeFalsy();
+      expect(result.message).toBe(`Expected element to have class 'baz', but had none`);
+
+      result = matcher.negativeCompare(tester.none, 'baz');
+      expect(result.pass).toBeTruthy();
+    });
+
+    it('should handle no element', () => {
+      // null element
+      let result = matcher.compare(null, 'baz');
+      expect(result.pass).toBeFalsy();
+      expect(result.message).toBe(`Expected to check class 'baz' on element, but element was falsy`);
+
+      result = matcher.negativeCompare(null, 'baz');
+      expect(result.pass).toBeTruthy();
```

Comment:
Bug here: it should be falsy. A test

    expect(null).not.toHaveClass('foo');

should fail.